examples/pytorch/graphsage/main.py

```diff
@@ -135,12 +122,13 @@ def train_func(config: Dict):
 
         losses_full.extend(losses)
         steps_in_epoch_trained = 0
-        if epoch % args.save_ckpt_by_epochs == 0:
-            save_checkpoint(model, logger, epoch, step, args)
+
+        if "model_path" in config:
+            save_checkpoint(model, logger, epoch, step, model_dir=config["model_path"])
```

Comment:
I've tried this before and it is not easy to work with, it was better to just use the utility functions. I forget what the issues were exactly.

Reply:
Please, write explicit issues you had with it. I don't see any new component in the main.py and don't understand what exactly we need to unit test.

Reply:
Can we remove the unit tests for graphsage?

Reply:
You are mixing two separate components in one:
1. Model.py still should be unit tested
2. We still need to run main.py in CI pipeline and check the output makes sense. Something like:
```bash
python main.py > log.txt
if ["($grep -ic 'Epoch 100: F1Score: 0.1' log.txt )" != "3"]
  then
    exit 3
fi
```

Reply:
I did not write the unit test, should I update it so that it only tests model.py? Ideally we just have main.py since if that works then the model works.

Reply:
> should I update it so that it only tests model.py?

yes

> main.py since if that works then the model works.

we also want to make sure if anyone runs examples, they work as expected.

Reply:
So to be clear - right now the unit test tests both the model and main.py by using the training loop in main.py and checking accuracy - but you want me to switch it so there are separate training loops in the unit test and main.py and we will test them separately?

Reply:
> So to be clear - right now the unit test tests both the model and main.py by using the training loop in main.py and checking accuracy

You are not using old data loaders anymore, so there is no point of sticking to testing model twice.

> but you want me to switch it so there are separate training loops in the unit test and main.py and we will test them separately?

You don't need a training loop in a unit test at all. Update it to check model queries graph/generates embeddings correctly, that's it. E2E can be verified in console.

> This seems like a lot of duplicate code and hacky unit test just to save a few lines in the example.

There will be way less code with this aproach.